### DISCUSSION TITLE
getting_started/introduction/learn_to_code_with_gdscript

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#40] [User-contributed notes]
# getting_started/introduction/learn_to_code_with_gdscript

In Godot, you can write code using the GDScript and C# programming languages. If you are new to programming, we recommend starting with GDScript because we designed it to be simpler than all-purpos...

https://docs.godotengine.org/en/stable/getting_started/introduction/learn_to_code_with_gdscript.html

**Comment by ZekeZDev** [comment]
Been programming for a bit and the GDScript From Zero didn't exist last time I was learning Godot but it was a really nice refresher for getting back into developing for Godot 4. Highly recommend doing it if you are new or coming back from a long break.

**Reply by Catley94** [reply]
I look forward to using GDScript, been recommended to use it since it's tightly integrated with the editor and easy to pick up and use.

**Reply by brianhsu** [reply]
I'm a experienced programmer, just not in gaming. I could say GDScript From Zero is worth checking if you are not familiar with programming. It gives you a really good start to understanding what programming is about and very important concept like error message is actually helping you. Definitely worth to check.

**Reply by Densket** [reply]
def gonna try his out, looks real nice

**Comment by laserbeak** [comment]
It felt like a python refresher, which was really nice.

**Comment by NeuroSeas** [comment]
I'm not new to programming, but neither am I a paragon of programming prowess. Thank you very much for the lovely refresher.

**Comment by gwpantazes** [comment]
"Learn GDScript From Zero" was absolutely delightful. I would highly recommend it to beginners in programming, not just for those learning Godot. For those more experienced, it's still a nice refresher on GDScript (and only GDScript, it doesn't really go into Godot libraries/nodes). I'm pretty much parroting what's already been said in previous comments, but I just had to comment because it was so well-explained and wonderful! I really enjoyed it, and I think it would be great to recommend as a starting point for kids even as early as elementary school who have learned their multiplication and/or just a beginnings of algebra. From that point on, it's off to the races! They can program with Godot and GDScript!

**Comment by CUBICinfinity** [comment]
If I opt to write my game in GDScript, How easy is it to make some parts use C# later?

**Reply by CUBICinfinity** [reply]
After reading more of the documentation, scripts are attached to nodes and it would appear that Godot is built to use both languages seamlessly. The object model also seems to make translation between languages not so difficult.

**Reply by Calinou** [reply]
It is possible to use GDScript and C# in the same project, as per the [Cross-language scripting](https://docs.godotengine.org/en/stable/tutorials/scripting/cross_language_scripting.html) documentation.

**Comment by marcoslaredo** [comment]
My only concern about the learning app (as well as the examples given at the use cases of the language) is that it is not ready for phones. It is not responsive to screen sizes, it is not ready for touch environments (rolling large text is a pain and do not trigger the virtual keyboard for texting).

**Comment by dev-expel** [comment]
The learning app doesn't actually load on my macbook.

**Reply by Calinou** [reply]
Which macOS version and browser are you using? Safari may have trouble, so try Chrome or Firefox.

**Comment by fabiansgz** [comment]
To this day I don't understand why if GDS and Python are similar they just didn't use Python and that's it. Can someone explain it?

**Reply by nasheomirro** [reply]
I think [this part of the docs](https://docs.godotengine.org/en/stable/about/faq.html#what-were-the-motivations-behind-creating-gdscript) can help answer that question.

**Comment by Cheeryo** [comment]
I just did the GDQuest tutorial and I am still impressed how great of an introduction this is. Thanks for the great work

**Comment by jangler** [comment]
It would be nice to link a "GDScript by example" or "GDScript cheatsheet" type resource here for folks who are already proficient programmers. The "learn GDScript from zero" resource is a poor fit for this group, especially since the information is gated behind a series of checkpoint questions, which makes it harder to skim and search.

**Reply by Calinou** [reply]
[GDScript reference](https://docs.godotengine.org/en/latest/tutorials/scripting/gdscript/gdscript_basics.html) is what you're looking for :slightly_smiling_face: 

There's also a third-party resource, [Learn X in Y Minutes](https://learnxinyminutes.com/gdscript/) which has a GDScript page.

**Comment by c3p0r2d2luke** [comment]
I wish that the repository was written in something Netlify-compatible. My school blocks GitHub.io and I wanted to do it :(

**Reply by Calinou** [reply]
There are downloadable native versions available on itch.io: https://gdquest.itch.io/learn-godot-gdscript

It should be possible to deploy the existing web export to a different host, since it's just a set of static files. However, this should be tracked on https://github.com/GDQuest/learn-gdscript/issues instead (please open an issue there).

**Comment by Fey-Morgan** [comment]
Hey guys, i'm very new to programming and this course is amazing. I really love learning how to use GDScript. It's very clear, clean and understandable for new folks. I just felt held by the hand to let me discover the world of coding. Thank you for this amazing ressources you give in the doc and via the course !

**Comment by AzziProto** [comment]
I usually don't go out of my way to leave comments on stuff, but I just gotta praise how solid this course is. I came into this with hardly any programming experience, and I really do feel like this helped me out a LOT. It was relatively simple and easy to understand, even for someone like me who is not that great at things like this.
Huge props and thanks to everyone who's worked on this, awesome stuff.
